code/game/objects/structures/crates_lockers/crates.dm

```diff
@@ -113,6 +113,18 @@
 	new /obj/item/bodypart/r_leg/robot/surplus(src)
 	new /obj/item/bodypart/r_leg/robot/surplus(src)
 
+//This is used for crates with randomly generated loot in 'code/modules/mining/abandoned_crates.dm' and 'code/modules/mining/lavaland/necropolis_chests.dm'
+/obj/structure/closet/crate/proc/pathorlist_to_loot(var/loot)

```

Comment:
```
/obj/structure/closet/crate/proc/pathorlist_to_loot(loot)
	if(!loot)
		return
	if(ispath(loot))
		new loot(src)
		return
	if(LAZYLEN(loot))
		for(var/i in loot)
			if(!ispath(i))
				continue
			for(var/ii in 1 to (loot[i] ? loot[i] : 1))//does i-in-loot have an associated value? if so use it, otherwise 1
				new i(src)
```